crates/api/src/chronicle_graphql/authorization.rs

```diff
@@ -0,0 +1,68 @@
+use base64::Engine;
+use jwtk::jwk::RemoteJwksVerifier;
+use serde_json::Value;
+use std::time::Duration;
+use thiserror::Error;
+
+#[derive(Debug, Error)]
+pub enum Error {
+    #[error("Base64 decoding failure")]
+    Base64 {
+        #[from]
+        source: base64::DecodeError,
+    },
+    #[error("JSON decoding failure")]
+    Json {
+        #[from]
+        source: serde_json::Error,
+    },
+    #[error("JWT validation failure")]
+    Jwks {
+        #[from]
+        source: jwtk::Error,
+    },
+    #[error("application error")]
+    Simple { message: String },
+}
+
+pub struct JwtChecker {
+    verifier: RemoteJwksVerifier,
+}
+
+impl JwtChecker {
+    pub fn new(jwks_uri: &str) -> Self {
+        Self {
+            verifier: RemoteJwksVerifier::new(jwks_uri.to_string(), None, Duration::from_secs(100)),
+        }
+    }
+
+    pub async fn verify_jwt(
+        &self,
+        token: &str,
+    ) -> Result<serde_json::map::Map<String, Value>, Error> {
+        let base64_engine = base64::engine::general_purpose::URL_SAFE_NO_PAD;
+
+        // JWT is composed of three base64-encoded components
+        let components = token
+            .split('.')
+            .map(|component| base64_engine.decode(component))
+            .collect::<Result<Vec<Vec<u8>>, base64::DecodeError>>()?;
+        if components.len() != 3 {
+            return Err(Error::Simple {
```

Comment:
Probably worth renaming this to Format rather than Simple?

Reply:
I was thinking of it as maybe a catchall for any other application-initiated errors that come up.

Reply:
Name them, it's much better to explicitly have separate types for errors or it ends up with the Go / rust-sawtooth-sdk pattern of error basically being String. Rust can be much more useful. It's also nice to be able to look at a modules Error type and see what kind of Bad Things can potentially happen. Error("Bad thing") doesn't have the same utility

Reply:
Current case in point - sawtooth sdk + zmq sending me UknownError("Unknown") which is plainly the result of this sort of thing somewhere (or possibly poor c interop)